### ISSUE TITLE
Queries

### ISSUE THREAD

**Issue by shahjaidev** [RUCAIBox/RecBole#1393]
Hi! 
I had a couple of questions, listing them here:

(i) I'm trying to train RecBole models on much larger data than the benchmark datasets. E.g. 2 million users x 2 million items, 15 million interactions
Is there any guide/suggestions for training on this scale? 

(ii) On a related note, since the Embedding Matrix for the above would be huge, does RecBole natively support sparse gradients (nn.Embedding(sparse=True) + SparseAdam

(iii) If one wants to provide a specific validation/test data instead of split from the training data, how to do that? 
Related to this, is there an option to provide a custom target item set that accompanies the validation set, during evaluation? (to make evaluation more efficient than full)

(iv) I wanted to confirm:
To obtain the final embeddings for each user, item:
Are these just the model's user_embedding.weight.numpy() , mapped from user_id to corresponding row?
That is, user_id 0 => Embedding is row 0 of the embedding matrix
user_id 1 => Row 1 and so on

(v) I found that for this data, LightGCN always runs out of GPU memory despite a small batch size. Any suggestions for working around this? 

Thanks :)

**Comment by AoiDragon** [comment]
@shahjaidev Hello, thanks for your attention to RecBole! 

For Q1: Support for parallel training on multiple GPUs is under development, you can expect to use this function around September.

For Q3: You can use **benchmark file** to load your own train/valid/test data separately, see [here](https://recbole.io/docs/user_guide/config/data_settings.html) for more details.

For Q4: No. I guess you are talking about embeddings in LightGCN. Actually, the user and item embeddings are stored in `user_all_embeddings` and `item_all_embeddings` after `foward()`. Then you can use `user_id` to get user embedding from the correspinding row of embedding matrix mentioned above.

For Q6: Yes, you are right. 

Hope these answers can help you. And I will reply to the rest of your questions as soon as possible.

**Comment by shahjaidev** [comment]
Thanks! 
For Q4: That is the case for LightGCN, but just to completely confirm, for BPR I can just convert the embedding weight matrix to numpy as user_embedding.weight.numpy().cpu() and that should be correctly mapped from user_id 0 => row 0 and so on.

**Comment by AoiDragon** [comment]
@shahjaidev Hello, here are the answers for the rest of your questions.

Q7: `AbstractDataLoader` in version 1.1.x  inherits from PyTorch DataLoader and you can set `num_workers` for your dataloader.

Q8: No. RecBole doesn't support learning rate scheduler in PyTorch at the moment.

Q9:  Ordinary negative sampling means randomly select items which didn't interact with users, while dynamic negative sampling will first choose those which get higher scores after each epoch. See `_neg_sampling()` in `recbole/data/dataloader/abstract_dataloader.py` for more details.

Q10: You can change the settings in `get_tensorboard()` in `recbole/utils/utils.py`.
